numpyro/handlers.py

```diff
@@ -192,7 +192,7 @@ def __init__(self, fn=None, guide_trace=None):
         super(replay, self).__init__(fn)
 
     def process_message(self, msg):
-        if msg['name'] in self.guide_trace and msg['type'] in ('sample', 'plate'):
+        if msg['type'] in ('sample', 'plate') and msg['name'] in self.guide_trace:
```

Comment:
I switch the order of two sides of `and` because some primitives such as `subsample` do not have `name` field.